pvlib/spectrum/mismatch.py

```diff
@@ -0,0 +1,213 @@
+"""
+The ``mismatch`` module provides functions for spectral mismatch calculations.
+"""
+
+import pvlib
+import numpy as np
+import pandas as pd
+from scipy.interpolate import interp1d
+import os
+
+
+def get_sample_sr(wavelength=None):
+    '''
+    Generate a generic smooth c-Si spectral response for tests and experiments.
+
+    Parameters
+    ----------
+    wavelength: 1-D sequence of numeric, optional
+        Wavelengths at which the spectral response should be interpolated.
+        By default the wavelengths are from 280 to 1200 in 5 nm intervals. [nm]
+
+    Returns
+    -------
+    sr: pandas.Series
+        The relative spectral response indexed by wavelength in nm. [-]
+
+    Notes
+    -----
+    This spectral response is based on measurements taken on a reference cell.
+    The measured data points have been adjusted by PV Performance Labs so that
+    standard cubic spline interpolation produces a curve without oscillations
+    as shown in [1]_.
+
+    References
+    ----------
+    .. [1] Driesse, Anton, and Stein, Joshua. "Global Normal Spectral
+       Irradiance in Albuquerque: a One-Year Open Dataset for PV Research".
+       United States 2020. https://doi.org/10.2172/1814068.
+       https://www.osti.gov/servlets/purl/1814068.
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Aug. 2022
+
+    SR_DATA = np.array([[ 290, 0.00],
+                        [ 350, 0.27],
+                        [ 400, 0.37],
+                        [ 500, 0.52],
+                        [ 650, 0.71],
+                        [ 800, 0.88],
+                        [ 900, 0.97],
+                        [ 950, 1.00],
+                        [1000, 0.93],
+                        [1050, 0.58],
+                        [1100, 0.21],
+                        [1150, 0.05],
+                        [1190, 0.00]]).transpose()
+
+    if wavelength is None:
+        resolution = 5.0
+        wavelength = np.arange(280, 1200 + resolution, resolution)
+
+    interpolator = interp1d(SR_DATA[0], SR_DATA[1],
+                            kind='cubic',
+                            bounds_error=False,
+                            fill_value=0.0,
+                            copy=False,
+                            assume_sorted=True)
+
+    sr = pd.Series(data=interpolator(wavelength), index=wavelength)
+
+    sr.index.name = 'wavelength'
+    sr.name = 'sr'
+
+    return sr
+
+
+def get_am15g(wavelength=None):
+    '''
+    Read the ASTM G173-03 AM1.5 global tilted spectrum, optionally interpolated
```

Comment:
According to a strict adherence to the G173-03 standard, I think the word "global" here (and elsewhere) should be "hemispherical". I recognize that "global" is nevertheless very common parlance, so perhaps just make a note of the synonymous usage here with "hemispherical".

From G173-03---

> 3.3.4.1 Discussion—For the special condition of a horizon-
tal plane the hemispherical solar irradiance is properly termed
global solar irradiance, E G . Incorrectly, global tilted, or total
global irradiance is often used to indicate hemispherical
irradiance for a tilted plane. In case of a sun-tracking receiver,
this hemispherical irradiance is commonly called global nor-
mal irradiance. The adjective global should refer only to
hemispherical solar radiation on a horizontal, not a tilted,
surface.

Reply:
Sure